src/osx/cocoa/dataview.mm

```diff
@@ -2910,16 +2912,45 @@ void UpdateWithRow(int row)
         }
     }
 
+    if ( attr.HasBackgroundColour() && [cell backgroundStyle] == NSBackgroundStyleLight )
+    {
+        // Only NSTextFieldCell provides a method for setting the background colour,
+        // so check that this method is available before using it
+        if ( [cell respondsToSelector:@selector(setBackgroundColor:)] &&
+                [cell respondsToSelector:@selector(backgroundColor)] )
+        {
+            if ( !data->GetOriginalBackgroundColour() )
+            {
+                // the cast to (untyped) id is safe because of the check above
+                data->SaveOriginalTextColour([(id)cell backgroundColor]);
+            }
+
+            colBack = attr.GetBackgroundColour().OSXGetNSColor();
+        }
+    }
+
+
     if ( !font )
         font = data->GetOriginalFont();
     if ( !colText )
         colText = data->GetOriginalTextColour();
+    if ( !colBack )
+        colBack = data->GetOriginalBackgroundColour();
 
     if ( font )
         [cell setFont:font];
 
     if ( colText )
         [(id)cell setTextColor:colText];
+
+    if ( colBack )
+    {
+        [(id)cell setDrawsBackground:true];
+        [(id)cell setBackgroundColor:colBack];
+    }
+    else
+        [(id)cell setDrawsBackground:false];
```

Comment:
Why is it safe to call `setDrawsBackground:` here? It doesn't seem obvious that `cell` is of the type which supports it... If this is correct, could you please add a comment explaining why is it so?

Reply:
Whoops, that should be guarded by a test for the `setDrawsBackground` function.